src/python-frontend/python_converter.cpp

```diff
@@ -0,0 +1,256 @@
+  {
+    return StatementType::FUNC_DEFINITION;
+  }
+  return StatementType::UNKNOWN;
+}
+
+std::string get_op(const std::string &op)
+{
+  auto it = operator_map.find(op);
+  if(it != operator_map.end())
+  {
+    return it->second;
+  }
+  return std::string();
+}
+
+typet get_typet(const std::string &ast_type)
+{
+  if(ast_type == "float")
+    return float_type();
+  if(ast_type == "int")
+    return int_type();
+  return empty_typet();
+}
+
+symbolt create_symbol(
+  const std::string &module,
+  const std::string &name,
+  const std::string id,
+  const locationt &location,
+  const typet &type)
+{
+  symbolt symbol;
+  symbol.mode = "Python";
+  symbol.module = module;
+  symbol.location = location;
+  symbol.type = type;
+  symbol.name = name;
+  symbol.id = id;
+  return symbol;
+}
+
+locationt get_location_from_decl(const nlohmann::json &ast_node)
+{
+  locationt location;
+  location.set_line(ast_node["target"]["lineno"].get<int>());
+  // TODO: Modify ast.py to include Python filename in the generated json
+  location.set_file("program.py"); // FIXME: This should be read from JSON
+  return location;
+}
+
+ExpressionType get_expression_type(const nlohmann::json &element)
+{
+  auto type = element["_type"];
+  if(type == "UnaryOp")
+  {
+    return ExpressionType::UNARY_OPERATION;
+  }
+  if(type == "BinOp")
+  {
+    return ExpressionType::BINARY_OPERATION;
+  }
+  if(type == "Constant")
+  {
+    return ExpressionType::LITERAL;
+  }
+  return ExpressionType::UNKNOWN;
+}
+
+exprt python_converter::get_binary_operator_expr(const nlohmann::json &element)
+{
+  exprt bin_expr(
+    get_op(element["op"]["_type"].get<std::string>()), current_element_type);
+
+  exprt lhs = get_expr(element["left"]);
+  exprt rhs = get_expr(element["right"]);
+
+  bin_expr.copy_to_operands(lhs, rhs);
+  return bin_expr;
+}
+
+exprt python_converter::get_unary_operator_expr(const nlohmann::json &element)
+{
+  exprt unary_expr(
+    get_op(element["op"]["_type"].get<std::string>()), current_element_type);
+
+  // get subexpr
+  exprt unary_sub = get_expr(element["operand"]);
+  unary_expr.operands().push_back(unary_sub);
+
+  return unary_expr;
+}
+
+exprt python_converter::get_expr(const nlohmann::json &element)
+{
+  exprt expr;
+  ExpressionType type = get_expression_type(element);
+
+  switch(type)
+  {
+  case ExpressionType::UNARY_OPERATION:
+  {
+    expr = get_unary_operator_expr(element);
+    break;
+  }
+  case ExpressionType::BINARY_OPERATION:
+  {
+    expr = get_binary_operator_expr(element);
+    break;
+  }
+  case ExpressionType::LITERAL:
+  {
+    auto value = element["value"];
+    if(element["value"].is_number_integer())
+    {
+      expr = from_integer(value.get<int>(), int_type());
+    }
+    break;
+  }
+  default:
+  {
+    assert(!"Unimplemented type in rule expression");
+  }
+  }
+
+  return expr;
+}
+
+symbolt python_converter::get_var_decl(const nlohmann::json &ast_node)
+{
+  // Get variable type
+  current_element_type =
```

Comment:
get_typet might return an empty type. Do we need to add an additional check after this one?

Reply:
empty_type represents void, not necessarily an error condition. So, it may not be necessary to handle it at this point.